Update documentation for ordinals (#175)
p2pderivatives/rust-dlc: docs/Ordinal.md

# Ordinal Support

> This library now supports the locking of an ordinal as part of the collateral of a DLC.

Reviewer: ```suggestion
This library supports the locking of an ordinal as part of the collateral of a DLC.
```

Notes:

> * An ordinal is inscribed into a single satoshi. However, that satoshi is often part of a larger UTXO. DLCs lock UTXOs, so please keep that context in mind.

Reviewer: I'm not an expert, but I don't think an ordinal is incribed. Some content can be "inscribed" into a single stats, but AFAIU an ordinal is the number associated with the sats.

Reviewer: Also I'm not sure what you're trying to say here tbh.

Author: Again I'm trying to give context for the whole effort. However, if you prefer to keep the document closer to as you had it before, I'll move some of this conversation outside to our own docs page. What do you think?

Reviewer: Yeah maybe that's better, I don't think this is the place where people should learn about ordinals tbh (I'd expect that someone who wants to use this feature of the library would already know about it).

Author: I agree. If someone's so deep in the rust-dlc code, they should have some background on the various technologies which brought them there.

> * In all cases, the gas fee to lock and later unlock the DLC is split between both parties.

Reviewer 2: gas fee is eth terminology.  I think you mean transaction fee.

> * For more information about DLCs, please see the main folder of this repo, the [`DLC-Link docs page`](https://docs.dlc.link), https://github.com/dlc-link, and the main https://github.com/discreetlogcontracts/dlcspecs specifications page.

Reviewer: I don't think that's relevant here, this is not supposed to be a standalone document.

Author: removed this whole section

> Resizing UTXOs
>

> > Please note, this package does not currently support the re-sizing of the UTXO containing the ordinal within the DLC. See [this section](#changing-postage) for more details

Reviewer: Why use `>` here?

Reviewer: Also since there is already a section about that under I feel this is redundant.

Author: OK, removed

## Supported types of DLCs

### DLC with only the ordinal
This package supports the use case when the offering party of the DLC locks the ordinal-containing UTXO into the DLC, and the counter-party (the acceptor) does not lock any collateral.

In this case, the outcome is essentially binary, either the ordinal containing UTXO goes to the offeror, or the acceptor.

### Ordinal with additional collateral
This package also supports the case when the offering party of the DLC locks the ordinal-contianing UTXO into the dlc, AND any amount of addition collateral is locked into the DLC by either party.

1. In this case, the outcome of the UTXO containing the ordinal will go to either party A or B, without resizing (as described earlier)

> 2. The remaining collateral locked by either, or both, parties can be split between the parties based on the standard DLC enumerated or numerical outcome guidelines.

Reviewer: I feel this could be summarized in a sentence saying that you can chose to attach an additional collateral or not. To me it's quite difficult to grasp as is.

Author: Well I was writing this from the perspective of someone who isn't fully versed in DLCs. So for that I was being more verbose and describing exactly what is happening in the DLC. Since there is no reference to Ordinals in the DLC specs anywhere, this can be used as the informational page. DLC Specs covers a lot of this high-level text. 

If you'd like however, I can shrink it back down, and find another place, perhaps on our dlc.link docs, to add this kind of less-technical review of ordinals and DLCs.

Reviewer: I think it's important to have information about how the library can be used, but somehow I'm not seeing these sentences as very helpful. I think what would be most helpful would be to have some [examples](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#examples) if you feel motivated.

Author: That's cool. As I try out the library on our dlc.link system i'll see if I can add some examples

> ### Locking an ordinal in the DLC

Reviewer: Should that be part of the `Supported types of DLC` section?

Author: Ah, probably not. Let me move stuff around a bit.

# Ordinal Support

This library supports the locking of an ordinal as part of the collateral of a DLC.

## Ordinals & additional collateral

The DLC can contain only 1 ordinal, either locked by the offering party or the accepting party.

Either of the participants can choose to include additional satoshis as collateral into the DLC, along with the ordinal.

## How it works

> In order to ensure that the ordinal does not get lost to fee, the DLC transactions for DLCs including an ordinal are created in the following way:

Reviewer: ```suggestion
In order to ensure that the ordinal gets properly assigned and not given to miners as fee, the DLC transactions for DLCs including an ordinal are created in the following way:
```

- The ordinal input is always set as the first one in the funding transaction,
- The funding output is always set as the first one in the funding transaction,
- The party getting the ordinal will always have its CET output in the first position.

> ### Locking an ordinal in the DLC

Reviewer: ```suggestion
### Locking an ordinal in a DLC
```
